src/rpc/blockchain.cpp

```diff
@@ -648,7 +648,14 @@ UniValue getblocksbatch(const UniValue& params, bool fHelp)
     try
     {
         // Have to do it this way, because the rpc param 0 must be left out of the special parameter handling in client.cpp.
-        nHeight = boost::lexical_cast<int>(params[0].get_str());
+        if (params[0].isNum())
+        {
+            nHeight = params[0].get_int();
+        }
+        else
+        {
+            nHeight = boost::lexical_cast<int>(params[0].get_str());
```

Comment:
shouldn't the try catch be moved to where the lexical_cast call is?

Reply:
It doesn't make a difference here, but yes, it is better form to invert the if and try.

Reply:
But both `get_int` and `get_str` can throw an `std::runtime_error` so you would still need a try-catch at the current level for the last catch statement.

Reply:
Yes. That is a different try-catch with a different class in the catch.

Reply:
Well I already included a (...) universal catch farther below. You probably should just move the if statement into the catch (...)

Something like
       if (params[0].isNum())
        {
            nHeight = params[0].get_int();
        }
        else
        {
            throw JSONRPCError(RPC_INVALID_PARAMETER, "Either a valid block number or block hash must be provided.");
        }

Reply:
Did you check the command line use case, because I think your placement of the if statement actually breaks the command line version where an "integer" block is passed in. The integer is treated as a string, because it is left out of the special fields type conversions, and therefore .get_int() should throw an exception. This will go to (...) and will result in an rpc error.

I think you have to do what I suggested above and move the get_int() into the catch (...)

I might be wrong. I haven't had the time to fool around with this.

Reply:
But if you put the if statement in the first catch and get_int throws an exception, it will not get caught by the universal catch.

Reply:
I did not test it from command line, but did test it in the graphical console, and tested it with JSON-RPC calls with both string and int types. I will try it a bit later.

Reply:
Just make sure it works for all cases and document the testing in the PR.

Reply:
You need to test blocknum as int, blocknum as string, hash for JSON direct RPC, blocknum and hash for rpc on the command line, and blocknum and hash for debug console.